src/ApplicationInsights.Kubernetes/Debuggings/InspectListener.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using Microsoft.Extensions.DiagnosticAdapter;
+
+namespace Microsoft.ApplicationInsights.Kubernetes.Debugging
+{
+    /// <summary>
+    /// The default listener of Application Insights for Kubernetes.
+    /// </summary>
+    public sealed class InspectListener
+    {
+        private InspectLevel _minimumLevel = InspectLevel.Warning;
+        private InspectListener() { }
+        /// <summary>
+        /// Gets the singleton instance of Application Insights for Kubernetes diagnostic source listener
+        /// </summary>
+        /// <returns></returns>
+        public static InspectListener Instance { get; } = new InspectListener();
+
+        /// <summary>
+        /// Sets the minimum level of the logs to show.
+        /// </summary>
+        public void SetMinimumLevel(InspectLevel newLevel) => _minimumLevel = newLevel;
+
+#pragma warning disable CA1822
+        /// <summary>
+        /// Invokes the message of critical.
+        /// </summary>
+        /// <param name="content"></param>
+        [DiagnosticName(nameof(InspectLevel.Critical))]
+        public void OnLogCritical(string content)
+        {
+            WriteLine(InspectLevel.Critical, content);
+        }
+
+        /// <summary>
+        /// Invokes the message of error.
+        /// </summary>
+        /// <param name="content"></param>
+        [DiagnosticName(nameof(InspectLevel.Error))]
+        public void OnLogError(string content)
+        {
+            WriteLine(InspectLevel.Error, content);
+        }
+
+        /// <summary>
+        /// Invokes the message of warning.
+        /// </summary>
+        /// <param name="content"></param>
+        [DiagnosticName(nameof(InspectLevel.Warning))]
+        public void OnLogWarning(string content)
+        {
+            WriteLine(InspectLevel.Warning, content);
+        }
+
+        /// <summary>
+        /// Invokes the message of information.
+        /// </summary>
+        [DiagnosticName(nameof(InspectLevel.Information))]
+        public void OnLogInfo(string content)
+        {
+            WriteLine(InspectLevel.Information, content);
+        }
+
+        /// <summary>
+        /// Invokes the message of debugging.
+        /// </summary>
+        [DiagnosticName(nameof(InspectLevel.Debug))]
+        public void OnLogDebug(string content)
+        {
+            WriteLine(InspectLevel.Debug, content);
+        }
+
+        /// <summary>
+        /// Invokes the message of tracing.
+        /// </summary>
+        [DiagnosticName(nameof(InspectLevel.Trace))]
+        public void OnLogTrace(string content)
+        {
+            WriteLine(InspectLevel.Trace, content);
+        }
+#pragma warning restore CA1822
+
+        private void WriteLine(InspectLevel level, string content)
+        {
+            if (level >= _minimumLevel)
```

Comment:
Honestly I am not sure if adding this listener to the library is beneficial. Just outputting the traces to console in unstructured format is probably not the right thing for most applications. Also this listener is very similar to the test listener you have created for tests, so if someone needs something like this, it is very easy to just copy the test listener code and tweak it as necessary.

Reply:
When you say not beneficial, I think you are coming from the end user's perspective. Then it is NOT, agree with you.

But this small piece is useful for the developers of AI.K8s. By just apply the listener, a simple console log / kubernetes pod log will be output and reveal what is happening of AI.K8s if something unexpected happens.

As a in-proc library, I have to swallow some generic exceptions to prevent crashing the target app. When that happens, the log helps.

Reply:
I don't want to add noise here. In my honest opinion, lib debugging is ok provided it is beneficial for the end users. If that is not the case, well I think we should re-consider it.